cmd/k8s-bigip-ctlr/main.go

```diff
@@ -369,6 +359,77 @@ func verifyArgs() error {
 	return nil
 }
 
+func getCredentials() error {
+	if len(*credsDir) > 0 {
+		var usr, pass, bigipURL string
+		var usrBytes, passBytes, urlBytes []byte
+		var err error
+		if strings.HasSuffix(*credsDir, "/") {
+			usr = *credsDir + "username"
+			pass = *credsDir + "password"
+			bigipURL = *credsDir + "url"
+		} else {
+			usr = *credsDir + "/username"
+			pass = *credsDir + "/password"
+			bigipURL = *credsDir + "/url"
+		}
+
+		usrBytes, err = ioutil.ReadFile(usr)
```

Comment:
Just an aside, this could be an anonymous function that takes a string and pointer called three times.